rest-server/core/src/main/scala/com/actionml/core/store/backends/MongoStore.scala

```diff
@@ -18,25 +18,74 @@
 package com.actionml.core.store.backends
 
 import java.time.{Instant, OffsetDateTime, ZoneOffset}
+import java.util.concurrent.TimeUnit
 
-import com.actionml.core.store.{DAO, Store}
+import com.actionml.core.store.indexes.annotations.Indexed
+import com.actionml.core.store.{DAO, Ordering, Store}
+import com.mongodb.client.model.IndexOptions
 import com.typesafe.scalalogging.LazyLogging
 import org.bson.codecs.configuration.{CodecProvider, CodecRegistries}
 import org.bson.codecs.{Codec, DecoderContext, EncoderContext}
 import org.bson.{BsonReader, BsonWriter}
+import org.mongodb.scala.bson.collection.immutable.Document
+import org.mongodb.scala.model.IndexModel
 import org.mongodb.scala.{MongoClient, MongoDatabase}
 
 import scala.concurrent.duration._
 import scala.concurrent.{Await, ExecutionContext, Future}
 import scala.reflect.ClassTag
+import scala.reflect.runtime.universe._
 
 
 class MongoStorage(db: MongoDatabase, codecs: List[CodecProvider]) extends Store with LazyLogging {
-  import scala.concurrent.ExecutionContext.Implicits.global
   import MongoStorage.codecRegistry
 
-  override def createDao[T](name: String)(implicit ct: ClassTag[T]): DAO[T] = {
-    new MongoDao[T](db.getCollection[T](name).withCodecRegistry(codecRegistry(codecs)(ct)))
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  override def createDao[T: TypeTag](name: String)(implicit ct: ClassTag[T]): DAO[T] = {
```

Comment:
again, TypeTag? Is this required to have access to the annotations?

Reply:
yes